MitoMathHelper.js

```diff
@@ -4,4 +4,82 @@
 
 const MitoMathHelper = class MitoMathHelper {
 
+
+
+    detectCollisionTime(circleA, velocityA, circleB, velocityB, interval){
+
+        let velocityIntA = this.multiplyPoint(velocityA, interval);
+        let velocityIntB = this.multiplyPoint(velocityB, interval);
+        let combineVelocity = this.addPoints(velocityIntA, this.invertPoint(velocityIntB));
+        let combineMagnitude = Math.hypot(combineVelocity[0], combineVelocity[1]);
+        //Early escape: Calculate bounds based on combine velocities
+        let bounds = new MitoCircle();
+        bounds.setPosition(circleA.getPosition()[0], circleA.getPosition()[1]);
+        bounds.setRadius(circleA.getRadius() +  combineMagnitude);
+        if(!this.detectCollisionCirCir(bounds, circleB)){
+            return null;
+        }
+        //Calculate closest point on combined velocities to circleB
+        let appliedVelocitiesToPositionA = this.addPoints(circleA.getPosition(), combineVelocity);
+        let closestPoint = this.getClosestPoint(circleA.getPosition(),
+            appliedVelocitiesToPositionA,
+            circleB.getPosition());
```

Comment:
circleB.getPosition(),
);

If it lets you, it depends on the language level were using.